### PR TITLE
Add Post and Get Categories

### PR DESCRIPTION
**Pull request by ahsanreal4**

### CODE REVIEW

File: frontend/src/hooks/useCreateCategories.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    setError(null);
+
+    const formData = {
+      name,
+    };
+
+    try {
+      const axiosInstance = await getAxiosInstance(true);
+      const response = await axiosInstance.post('/api/categories', formData);
+      getAuthToken()
```
Comment (ahsanreal4) [inline frontend/src/hooks/useCreateCategories.jsx]: why is getAuthToken() called here?

File: frontend/src/hooks/useCreatePosts.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    const formData = {
+      title,
+      content,
+      categoryId,
+    };
+
+    try {
+      const axiosInstance = await getAxiosInstance(true);
+      const response = await axiosInstance.post('/api/posts', formData);
+      getAuthToken()
```
Comment (ahsanreal4) [inline frontend/src/hooks/useCreatePosts.jsx]: Also here why is getAuthToken() called ?

File: frontend/src/hooks/useFetchCategories.jsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+function useFetchCategories() {
+  const [categoriesData, setCategoriesData] = useState([]);
+  const dispatch = useDispatch(); 
+
+  const fetchCategoriesData = async () => {
+    try {
+      const axiosInstance = await getAxiosInstance(true);
+      const response = await axiosInstance.get('/api/categories');
+      setCategoriesData(response.data); 
+      getAuthToken()
```
Comment (ahsanreal4) [inline frontend/src/hooks/useFetchCategories.jsx]: same here getAuthToken() called without any use case

File: frontend/src/pages/Dashboard.jsx [hunk 4]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
 import { FiPlus } from 'react-icons/fi';
 import useFetchUsers from '../hooks/useFetchUsers';
+import { useNavigate } from 'react-router-dom';
 function Dashboard() {
   useFetchUsers();
-
+  const navigate = useNavigate()
+  const handleNavigate = () =>{
+    navigate("/posts")
```
Comment (ahsanreal4) [inline frontend/src/pages/Dashboard.jsx]: why you are passing string directly here ? 
Use PAGES.POSTS to navigate

File: frontend/src/pages/Posts.jsx
Comment (ahsanreal4) [inline frontend/src/pages/Posts.jsx:1]: Seperate the html a little bit into components so the code is easier to read like CategoriesDropdown, CatogeryInput and TextEditor

File: frontend/src/hooks/useFetchCategories.jsx
Comment (ahsanreal4) [inline frontend/src/hooks/useFetchCategories.jsx:1]: This hook is against the SRP ( Single Responsibility Principle ). This hook should only be used to get categories and return them. Setting redux state in this hook makes this hook non-generic. Set the state of redux else where.

File: frontend/src/pages/Posts.jsx
Comment (Faizan707) [reply frontend/src/pages/Posts.jsx:1]: it looks fine 😜

File: frontend/src/hooks/useFetchCategories.jsx
Comment (ahsanreal4) [reply frontend/src/hooks/useFetchCategories.jsx:1]: this hook is still not correct. return categories response from the fetchCategoriesData function.
And dispatch the redux state outside of the hook's function.

**Review by ahsanreal4 — COMMENTED** [review]
done

**Pull request merged**